### PR TITLE
GIBCT keyword search

### PR DESCRIPTION
**Pull request by bshyong**
Resolves https://github.com/department-of-veterans-affairs/vets.gov-team/issues/1536

Users can press enter or select a result from the autocomplete suggestions to trigger a new search. Additionally, the search terms are reflected in the URL and will be persisted upon a page refresh to the same URL.

### CODE REVIEW

File: src/js/gi/components/search/KeywordSearch.jsx [hunk 1]
```diff
@@ -15,6 +15,33 @@ export class KeywordSearch extends React.Component {
...
+  componentDidMount() {
+    const searchQuery = this.props.location && this.props.location.query;
+    if (searchQuery) {
+      this.handleChange(null, { newValue: searchQuery.name, method: 'enter' });
+    }
+  }
+
+  onKeyUp(e) {
+    const { onFilterChange, autocomplete } = this.props;
+    if ((e.which || e.keyCode || 0) === 13) {
```
Comment (U-DON) [inline src/js/gi/components/search/KeywordSearch.jsx]: Is the `|| 0` here necessary?

File: src/js/gi/components/search/KeywordSearch.jsx [hunk 2]
```diff
@@ -15,6 +15,33 @@ export class KeywordSearch extends React.Component {
...
+      onFilterChange('name', autocomplete.searchTerm);
+    }
+  }
+
+  handleChange(event, data) {
+    this.props.updateAutocompleteSearchTerm(data.newValue);
+  }
+
+  handleSuggestionSelected(event, data) {
+    this.props.onFilterChange('name', data.suggestionValue);
```
Comment (U-DON) [inline src/js/gi/components/search/KeywordSearch.jsx]: Any issues when `onFilterChange` isn't provided, like on the `LandingPage`? Presumably not, but just wanted to check.

**Review by U-DON — APPROVED** [review]
LGTM. Had some minor comments.

File: src/js/gi/components/search/KeywordSearch.jsx (hunk 2 above)
Comment (bshyong) [inline src/js/gi/components/search/KeywordSearch.jsx]: Good catch; I set a no-op as a default prop for this to prevent any errors on the LandingPage.

**Pull request merged**